onboarding.md

```diff
@@ -0,0 +1,591 @@
+- `transfer-file`: contains all the code necessary for the file transfer
+
+### Create a control REST API
+
+We will need some way to interact with the consumer, a communication protocol of sorts. That is what we call
+outward-facing communication. In this example we will communicate with the consumer via simple command line tools
+like `cURL`, but it is easy to imagine some other much more complicated control system to interact with the (consumer)
+connector. Thus, using Jakarta, we must create an API controller the same way we created our health endpoint a few
+chapters back. In fact, we can re-use and
+improve [that controller](samples/04-file-transfer/api/src/main/java/org/eclipse/dataspaceconnector/extensions/api/ConsumerApiController.java)
+(code omitted here for brevity).
+
+Again, like before, the controller is instantiated and registered in an extension which we aptly
+name `ApiEndpointExtension.java`.
+
+### Create the "provider" connector
+
+The provider is the one "owning" the data. That means, the consumer sends a request to the provider (over IDS), who then
+performs the file transfer. The provider must maintain an internal list of assets that are available for transfer, the
+so-called "catalog". For the sake of simplicity we use an in-memory catalog and pre-fill it with just one single class:
+
+```java
+// in FileTransferExtension.java
+@Override
+public void initialize(ServiceExtensionContext context){
+        // ...
+        registerDataEntries(context);
+        // ...
+        }
+
+private void registerDataEntries(ServiceExtensionContext context){
+        var metadataStore=context.getService(MetadataStore.class);
+
+        GenericDataCatalogEntry file1=GenericDataCatalogEntry.Builder.newInstance()
+        .property("type","File")
+        .property("path","/path/to/assets/")
+        .property("filename","test-document.txt")
+        .build();
+
+        DataEntry entry1=DataEntry.Builder.newInstance().id("test-document").policyId(USE_EU_POLICY).catalogEntry(file1).build();
+        metadataStore.save(entry1);
+        }
+```
+
+This adds a `GenericDataCatalogEntry` to the `MetadataStore` (which is the catalog). Or, in other words, your provider
+now "hosts" one file named `test-documents.txt` located in the path `/path/to/assets/` on your development machine. It
+makes it available for transfer under its `id` `"test-document"`. While it makes sense to have some sort of similarity
+between file name and id, it is by no means mandatory.
+> **Please adjust this path as necessary to match your local environment!**
+
+Please also note that the registering of Policies is omitted from this document for clarity.
+
+The `FileTransferExtension` also registers a `FileTransferFlowController`, which is basically a glorified `cp` process
+with some additional checking Please note that the provider connector does *not* add the `api` module, so that means
+that no REST API is offered by the provider!
```

Comment:
but once again something that will be changed soon - especially if we want to send messages back and forth async

Reply:
sure, but the guide aims to portray the current state of development. ideally the guide gets updated, whenever we update any of those core features